File: android/ValleCOM/app/src/main/java/com/valleapp/valleCOM/adaptadores/AdaptadorAutorias.java
package com.valleapp.valleCOM.adaptadores;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageButton;
import android.widget.TextView;

import com.valleapp.valleCOM.R;

import org.json.JSONObject;

import java.util.List;

public class AdaptadorAutorias extends ArrayAdapter<JSONObject> {

    List<JSONObject> values;
    Context cx;

    public AdaptadorAutorias(@NonNull Context context, List<JSONObject> obj) {
        super(context, R.layout.linea_autoria, obj);
        this.values = obj;
        this.cx = context;
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        LayoutInflater inflater = (LayoutInflater) cx
                .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View v = inflater.inflate(R.layout.linea_autoria, parent, false);
        try {
            JSONObject o = values.get(position);
            TextView t = v.findViewById(R.id.txt_mensaje_autoria);
            ImageButton b = v.findViewById(R.id.btn_autorizar);
            if(!o.getString("tipo").equals("informacion")) {
                b.setTag(o);
            }else{
                b.setVisibility(View.GONE);
            }
            ImageButton bc = v.findViewById(R.id.btn_denegar);
            bc.setTag(o);
            t.setText(o.getString("mensaje"));

        }catch (Exception e){
            e.printStackTrace();
        }

        return v;
    }
}

File: android/ValleTPV/ValleCOM/src/main/java/com/valleapp/vallecom/adaptadores/AdaptadorAutorias.kt
package com.valleapp.vallecom.adaptadores

import android.content.Context
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.valleapp.vallecom.activitys.Autorias
import com.valleapp.valletpv.R
import org.json.JSONObject

class AdaptadorAutorias(
    private val context: Context,
    private val values: List<JSONObject>
) : RecyclerView.Adapter<AdaptadorAutorias.AutoriasViewHolder>() {

    inner class AutoriasViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val mensajeTextView: TextView = view.findViewById(R.id.txt_mensaje_autoria)
        val autorizarButton: ImageButton = view.findViewById(R.id.btn_autorizar)
        val denegarButton: ImageButton = view.findViewById(R.id.btn_denegar)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AutoriasViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.linea_autoria, parent, false)
        return AutoriasViewHolder(view)
    }

    override fun onBindViewHolder(holder: AutoriasViewHolder, position: Int) {
        try {
            val item = values[position]
            holder.mensajeTextView.text = item.getString("mensaje")

            if (item.getString("tipo") != "informacion") {
                holder.autorizarButton.tag = item
                holder.autorizarButton.visibility = View.VISIBLE
                holder.autorizarButton.setOnClickListener { view ->
                    // Lógica para manejar el clic del botón autorizar
                    (context as? Autorias)?.sendAutorizacion(view)
                }
            } else {
                holder.autorizarButton.visibility = View.GONE
            }

            holder.denegarButton.tag = item
            holder.denegarButton.setOnClickListener { view ->
                // Lógica para manejar el clic del botón denegar
                (context as? Autorias)?.sendCancelacion(view)
            }
        } catch (e: Exception) {
            Log.e("AdaptadorAutorias", "Error al enlazar datos: ${e.message}", e)
        }
    }

    override fun getItemCount(): Int {
        return values.size
    }
}
